src/spesh/optimize.c

```diff
@@ -1149,8 +1149,6 @@ static void optimize_string_equality(MVMThreadContext *tc, MVMSpeshGraph *g, MVM
     MVMSpeshFacts *b_facts = MVM_spesh_get_facts(tc, g, ins->operands[2]);
     MVMuint8 was_eq = 0;
 
-    return;
-
```

Comment:
This may well have been done to disable an optimization that is broken, so this is a bit risky

Reply:
The `return` was in the original commit (9d8000c) that introduced `optimize_string_equality`, so I'm not sure. But a spectest does pass with the `return` removed.